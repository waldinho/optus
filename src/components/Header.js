import React from 'react';
import styled from 'styled-components';

import * as style from '../styleVars/variables.js';

const Header = () => (
    <Heading>
        <img className='logo' src='https://1.cdn.optusdigital.com/content/dam/optus/images/about-us/media-centre/multimedia/logos/OPTUS_Yellow_sRGB_RELEASE_03_310316.png/renditions/version-1506405814053/original.png' alt='Optus' title='Optus'/>
    </Heading>
) 

const Heading = styled.div`
    background-color: ${style.black};
    min-height: 6vh;
    ${style.column}
    align-items: center;
    justify-content: center;
    font-size: calc(${style.sm} + 2vmin);
    color: ${style.yellow};
    min-width: 335px;
    position: fixed;
    top: 0;
    width: -webkit-fill-available;
    z-index: 2;
    border-bottom: 1px solid ${style.black};
    .logo {
        width: 80px;
    }
`

export default Header