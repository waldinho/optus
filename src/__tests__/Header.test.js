import React from 'react';
import ReactDOM from 'react-dom';
import Header from '../components/Header';
import { shallow } from 'enzyme';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Header />, div);
  ReactDOM.unmountComponentAtNode(div);
});

describe('<Header /> with no props', () => {
    const container = shallow(<Header />);
    it('should match the snapshot', () => {
      expect(container.html()).toMatchSnapshot();
      expect(container.find('img').prop('src')).toEqual('https://1.cdn.optusdigital.com/content/dam/optus/images/about-us/media-centre/multimedia/logos/OPTUS_Yellow_sRGB_RELEASE_03_310316.png/renditions/version-1506405814053/original.png');
      expect(container.find('img').prop('alt')).toEqual('Optus');
      expect(container.find('img').prop('title')).toEqual('Optus');
      expect(container.find('img').prop('className')).toEqual('logo');
    })
})
