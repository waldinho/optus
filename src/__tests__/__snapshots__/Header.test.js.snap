// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> with no props should match the snapshot 1`] = `"<div class=\\"sc-AxjAm bsWaWi\\"><img class=\\"logo\\" src=\\"https://1.cdn.optusdigital.com/content/dam/optus/images/about-us/media-centre/multimedia/logos/OPTUS_Yellow_sRGB_RELEASE_03_310316.png/renditions/version-1506405814053/original.png\\" alt=\\"Optus\\" title=\\"Optus\\"/></div>"`;
